src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -758,6 +758,8 @@ void RaftReplDev::handle_commit(repl_req_ptr_t rreq, bool recovery) {
 
     // Remove the request from repl_key map.
     m_repl_key_req_map.erase(rreq->rkey());
+    // Remove the request from lsn map.
+    m_state_machine->unlink_lsn_to_req(rreq->lsn());
```

Comment:
if we do this , we will lose the entry cache in leader.
what`s the problem if remove this.
https://github.com/eBay/HomeStore/blob/72a355e80a8e8368b555637e491840bc6307eb77/src/lib/replication/repl_dev/raft_repl_dev.cpp#L1092

can the reqs in the leader be removed as those in follower when they are expired?

Reply:
this is not used anywhere except in `end_of_append_batch`  ATM.

Reply:
I think the `entry cache` is separated into two part - log and data.
Log is on logstore, as discussed in your logstore PR if the cache you add can be extend and cache last N records, regardless of flush.

Data is on the path of `RaftReplDev::on_fetch_data_received`, right now it just read from drive based on blkid. As https://github.com/eBay/HomeStore/issues/414 stated this approach is wrong after we introduce GC. If the fetch need to go through Index it goes through HO,  I think we may have a unified cache in HO which can serve customer IO as well.